build-tools/packages/build-cli/src/library/repoPolicyCheck/npmPackages.ts

```diff
@@ -505,6 +509,211 @@ function getPreferredScriptLine(scriptLine: string): string {
+}
+
+/**
+ * Determine missing elements to properly lint API exports.
+ *
+ * @remarks Does not check that api-extractor config files have linting enabled.
+ *
+ * @param packageJson - package.json contents
+ * @param dir - directory of package.json
+ * @returns record of missing requirements or with unexpected values
+ */
+async function getApiLintElementsMissing(
+	packageJson: Readonly<PackageJson>,
+	dir: string,
+): Promise<{
+	scriptEntries: ScriptEntry[];
+	configFiles: Map<string, string>;
+	devDependencies: string[];
+	targetsImpacted: Set<string>;
+}> {
+	const scriptEntries: ScriptEntry[] = [];
+	const configFiles = new Map<string, string>();
+	const devDependencies: string[] = [];
+	const targetsImpacted = new Set<string>();
+	const missing = { scriptEntries, configFiles, devDependencies, targetsImpacted };
+
+	const exportsField = packageJson.exports;
+	if (exportsField === undefined) {
+		return missing;
+	}
+
+	const { mapTypesPathToExportPaths } = queryTypesResolutionPathsFromPackageExports(
+		packageJson,
+		new Map([[/\.d\.ts$/, undefined]]),
+		{ node10TypeCompat: false, onlyFirstMatches: false },
+	);
+
+	const needsLinted = new Map<string, string>();
+	let bundleLintTarget: string | undefined;
+	for (const [relPath, exports] of mapTypesPathToExportPaths.entries()) {
+		const onlyRequire = exports.every((e) => e.conditions.includes("require"));
+		const onlyImport = exports.every((e) => e.conditions.includes("import"));
+		const skew = onlyRequire ? "cjs:" : onlyImport ? "esm:" : "";
+		if (exports.some((e) => !e.exportPath.startsWith("./internal"))) {
+			const existingSkew = needsLinted.get(relPath);
+			if (existingSkew === undefined) {
+				needsLinted.set(relPath, skew);
+			} else if (existingSkew !== skew) {
+				needsLinted.set(relPath, "");
+			}
+		} else if (exports.some((e) => e.exportPath === "./internal")) {
+			// ./internal export should be checked for cross repo consistency.
+			// Only one file needs to be checked for this. Prefer export that
+			// is not 'require' restricted.
+			// eslint-disable-next-line unicorn/no-lonely-if
+			if (bundleLintTarget === undefined || !onlyRequire) {
+				bundleLintTarget = relPath;
+			}
+		}
+	}
+	if (needsLinted.size === 0) {
+		// No files need linting or the only file that should be linted is internal,
+		// which is not checked by policy
+		return missing;
+	}
+
+	// -------------------------------------------------------------------------
+	// There are files that need linting.
+
+	// Make sure the package.json has the check:exports script that runs others.
+	const checkExports = packageJson.scripts?.["check:exports"];
+	if (checkExports !== 'concurrently "npm:check:exports:*"') {
+		scriptEntries.push({
+			name: "check:exports",
+			commandLine: 'concurrently "npm:check:exports:*"',
+		});
+	}
+
+	// Make sure `concurrently` is available.
+	if (packageJson.devDependencies?.concurrently === undefined) {
+		devDependencies.push("concurrently");
```

Comment:
There's already a policy that checks that dependencies exist for commands used in the package scripts. Why is that policy not catching this?

Reply:
It very well could (what is the policy?), but if this introducing the dependency, then it should fix it, I think, rather than requiring a second pass of fixing.

Reply:
> if this introducing the dependency, then it should fix it, I think, rather than requiring a second pass of fixing.

Good point. The policy I'm referring to is called npm-package-json-script-dep and is configured here. Looks like concurrently isn't there but it should be. We ca fix that separately.

https://github.com/microsoft/FluidFramework/blob/c6293d4b833da067c9abbd447eb9483fc8425be9/fluidBuild.config.cjs#L413-L436

Reply:
PR to fix the concurrently omission: https://github.com/microsoft/FluidFramework/pull/21259

Reply:
Unfortunately npm-package-json-script-dep doesn't have a resolver. I guess it can't pick a version. I think there is something else that cares about consistency. Maybe those two should get together to have a resolver.

Reply:
PR to add concurrently support to fluid-build: #21262